engine/renderer/private/gpu_scene.cpp

```diff
@@ -39,33 +41,59 @@ GPUScene::~GPUScene()
     vkContext->Device().destroy(_objectInstancesDescriptorSetLayout);
 }
 
-void GPUScene::Update(const SceneDescription& scene, uint32_t frameIndex)
+void GPUScene::Update(uint32_t frameIndex)
 {
-    UpdateSceneData(scene, frameIndex);
+    UpdateSceneData(frameIndex);
     UpdateObjectInstancesData(frameIndex);
     WriteDraws(frameIndex);
 }
 
-void GPUScene::UpdateSceneData(const SceneDescription& scene, uint32_t frameIndex)
+void GPUScene::UpdateSceneData(uint32_t frameIndex)
 {
     SceneData sceneData {};
 
-    const DirectionalLight& light = scene.directionalLight;
+    auto directionalLightView = _ecs->registry.view<DirectionalLightComponent, TransformComponent>();
+    bool directionalLightIsSet = false;
 
-    // Calculate light direction from Euler rotation
-    glm::vec3 direction = glm::vec3(
-        cos(light.camera.eulerRotation.y) * cos(light.camera.eulerRotation.x),
-        sin(light.camera.eulerRotation.x),
-        sin(light.camera.eulerRotation.y) * cos(light.camera.eulerRotation.x));
-
-    const glm::mat4 lightView = glm::lookAt(light.camera.position, light.camera.position - direction, glm::vec3(0, 1, 0));
-    glm::mat4 depthProjectionMatrix = glm::ortho(-light.camera.orthographicSize, light.camera.orthographicSize, -light.camera.orthographicSize, light.camera.orthographicSize, light.camera.nearPlane, light.camera.farPlane);
-    depthProjectionMatrix[1][1] *= -1;
-
-    DirectionalLightData& directionalLightData = sceneData.directionalLight;
-    directionalLightData.lightVP = depthProjectionMatrix * lightView;
-    directionalLightData.depthBiasMVP = DirectionalLight::BIAS_MATRIX * directionalLightData.lightVP;
-    directionalLightData.direction = glm::vec4(direction, light.shadowBias);
+    directionalLightView.each([this, &sceneData, &directionalLightIsSet](const auto directionalLightComponent, const auto transformComponent)
```

Comment:
hmm can't we create a view and then use a for loop? I think it will look a little better instead of a lambda, and we can use a counter variable, since later we might support more then one directional light